### ISSUE TITLE
False positive `invalid-overridden-method` when overridding `Enum.value`

### ISSUE THREAD

**Issue by e-gebes** [pylint-dev/pylint#9905]
### Bug description

```python
import enum


class Thing(enum.Enum):
    A = enum.auto()

    @property
    def value(self):
        return "thing"
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
8:4: W0236: Method 'value' was expected to be 'method', found it instead as 'property' (invalid-overridden-method)
```

### Expected behavior

I tried to lint some legacy code and thought this might be a false positive.
I would expect no warning, or how could I otherwise override `value` from the Enum class? (not saying this is a good idea in general)

Maybe this is related to decorators: https://github.com/pylint-dev/pylint/issues/8939 and the CPython implementation also has a decorator which is similar to a property:

https://github.com/python/cpython/blob/3.10/Lib/enum.py#L802

```py
    @DynamicClassAttribute
    def value(self):
        """The value of the Enum member."""
        return self._value_
```

### Pylint version

```shell
pylint 2.17.7
astroid 2.15.8
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]
```
